I18n implementation

NeroxTGC opened the pull request:
## i18n implementation

- Browser Language detector
- Fallback/default language: EN
- Available languages: EN/ES
- Added `LanguageSelector` to the `NavBar`

![image](https://github.com/user-attachments/assets/698d0948-4b99-4100-b345-da1a71fabece)
![image](https://github.com/user-attachments/assets/de6e7fbe-3f38-4102-a047-697083ebd0b7)

### Translated:
- Landing Page
- Pricing Page
- Account Page
- `FileUploadPage`
- `DemoAppPage`
- `NavBar` component
- `UserMenuItems` component
- `Footer `component
---

## Contributor Checklist 

> Make sure to do the following steps if they are applicable to your PR:

**(not done)**

- [ ] **Update e2e tests**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/template/e2e-tests](/template/e2e-tests) also.
- [ ] **Update demo app**: If you changed the [/template/app](/template/app), then make sure to do any neccessary updates to [/opensaas-sh/app_diff](/opensaas-sh/app_diff) also. Check [/opensaas-sh/README.md](/opensaas-sh/README.md) for details.
- [ ] **Update docs**: If needed, update the [/opensaas-sh/blog/src/content/docs](/opensaas-sh/blog/src/content/docs).

NeroxTGC committed:
i18n implementation

i18n implemented:
- Languages: EN/ES
- User language detection
- Default language: EN

Translated pages:
Landing Page, Pricing, Account, FileUploadPage, DemoAppPage

NeroxTGC committed:
fixed

Martinsos commented:
Cool stuff @NeroxTGC !
What do you say @vincanger ? Makes sense to me (I haven't checked the code, but as a concept). I guess the question is if it is possibly too much to put in the template, but I guess it might be fine, people have asked for it quite often? If not we can potentially move it into a guide instead hm.

NeroxTGC committed:
Saving option to local storage (bug)

This way, even when the page is refreshed, it will:

>First check localStorage for a saved language
>If found, use that language
>If not found, save the current language (or 'en' as fallback) to localStorage

This ensures the language preference persists across page refreshes.
